src/charts/helpers/text.js

```diff
@@ -71,76 +71,75 @@ define(function(require) {
      * to configure a y offset for entries with multiple rows
-     * @param  {D3Sekectuib} text       d3 text element
+     * @param  {D3Selection} text       D3 text element
      * @param  {Number} width           Width of the container where the text needs to wrap on
-     * @param  {Number} xpos            number passed to determine the x offset
-     * @param  {Number} limit           number of lines before an ellipses is added and the rest of the text is cut off
+     * @param  {Number} [xpos=0]        Number passed to determine the x offset
+     * @param  {Number} [limit=2]       Number of lines before an ellipses is added and the rest of the text is cut off
      *
      * REF: http://bl.ocks.org/mbostock/7555321
      * More discussions on https://github.com/mbostock/d3/issues/1642
      * @return {void}
      */
     const wrapTextWithEllipses = function(text, width, xpos=0, limit=2) {
+        text.each(function() {
+            var words,
+                word,
+                line,
+                lineNumber,
+                lineHeight,
+                y,
+                dy,
+                tspan;
+
+            text = d3Selection.select(this);
+
+            words = text.text().split(/\s+/).reverse();
+            line = [];
+            lineNumber = 0;
+            lineHeight = 1.2;
+            y = text.attr('y');
+            dy = parseFloat(text.attr('dy'));
+            tspan = text
+                .text(null)
+                .append('tspan')
+                .attr('x', xpos)
+                .attr('y', y)
+                .attr('dy', dy + 'em');
+
+            while ((word = words.pop())) {
+                line.push(word);
+                tspan.text(line.join(' '));
 
-            text.each(function() {
-                var words,
-                    word,
-                    line,
-                    lineNumber,
-                    lineHeight,
-                    y,
-                    dy,
-                    tspan;
-
-                text = d3Selection.select(this);
-
-                words = text.text().split(/\s+/).reverse();
-                line = [];
-                lineNumber = 0;
-                lineHeight = 1.2;
-                y = text.attr('y');
-                dy = parseFloat(text.attr('dy'));
-                tspan = text
-                    .text(null)
-                    .append('tspan')
-                    .attr('x', xpos)
-                    .attr('y', y)
-                    .attr('dy', dy + 'em');
-
-                while ((word = words.pop())) {
-                    line.push(word);
+                if (tspan.node() && tspan.node().getComputedTextLength() > width) {
+                    line.pop();
                     tspan.text(line.join(' '));
 
-                    if (tspan.node() && tspan.node().getComputedTextLength() > width) {
-                        line.pop();
+                    if (lineNumber < limit - 1) {
```

Comment:
Shouldn't `(lineNumber < limit)` work instead? This looks like one of the lines will be left out.

Reply:
Ha, no idea either!

I can add it to my list of stuff to investigate... but I know I will never get there!